tests/test_jsvalidation.py

```diff
@@ -0,0 +1,419 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of DoSchema.
+# Copyright (C) 2016 CERN.
+#
+# DoSchema is free software; you can redistribute it
+# and/or modify it under the terms of the GNU General Public License as
+# published by the Free Software Foundation; either version 2 of the
+# License, or (at your option) any later version.
+#
+# DoSchema is distributed in the hope that it will be
+# useful, but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with DoSchema; if not, write to the
+# Free Software Foundation, Inc., 59 Temple Place, Suite 330, Boston,
+# MA 02111-1307, USA.
+#
+# In applying this license, CERN does not
+# waive the privileges and immunities granted to it by virtue of its status
+# as an Intergovernmental Organization or submit itself to any jurisdiction.
+
+import jsonschema
+import pytest
+
+from doschema.errors import JSONSchemaCompatibilityException, \
+    JSONSchemaDependenciesException
+from doschema.jsvalidation import JSONSchemaValidation
+
+
+def test_special_field_fail():
+    v1 = {
+        "type": "object",
+        "oneOf": [{
+            "type": "object",
+            "properties": {
+                "field_A": {"type": "string"}
+            }
+        }, {
+            "type": "object",
+            "properties": {
+                "field_A": {"type": "integer"}
+            }
+        }]
+    }
+    obj = JSONSchemaValidation()
+    with pytest.raises(JSONSchemaCompatibilityException):
+        obj.start(v1, 'first')
+
```

Comment:
Here are a few tests I think won't pass:
```json
{
    "type":"object",
    "properties":{  
        "root_field":{  
            "type":"array",
            "items":{  
                "type":"object",
                "properties":{  
                    "sub_field":{  
                        "type":"string"
                    }
                }
            }
        }
    }
}
```
and
```json
{
    "type":"object",
    "properties":{  
        "root_field":{  
            "type":"array",
            "items":{  
                "type":"object",
                "properties":{  
                    "sub_field":{  
                        "type":"integer"
                    }
                }
            }
        }
    }
}
```
Should conflict


This schema should conflict with itself:
```json
{  
    "allOf":[  
        {  
            "type":"object",
            "properties":{  
                "my_field":{  
                    "type":"string"
                }
            }
        },
        {  
            "type":"object",
            "dependencies":{  
                "my_field":{  
                    "type":"object",
                    "properties":{  
                        "my_field":{  
                            "type":"integer"
                        }
                    }
                }
            }
        }
    ]
}
```

I think this one would be handled strangely by this code (yes it is ugly):
```json
{
	"type": "object",
	"properties": {
		"properties": {
			"type": "string"
		},
		"type": {
			"type": "string"
		},
		"items": {
			"type": "array",
			"items": {
				"type": "string"
			}
		},
		"dependencies": {
		    "properties": {
			    "dependencies": "integer",
			},
			"dependencies": {
				"dependencies": {
					"type": "object",
					"properties": {
						"dependencies": {
							"type": "string"
						}
					}
				}
			}
		}
	}
}
```
Even if it is ugly, most of it is valid. We should only raise an exception because the field `dependencies.dependencies` is once declared as `string` and once as `integer`.